src/App.js

```diff
@@ -2,13 +2,68 @@ import React from "react";
 import "./App.css";
 
 class App extends React.Component {
-  render() {
-    return (
-      <main>
-        <p>React State Lab</p>
-      </main>
-    );
-  }
+	constructor() {
+		super();
+		this.state = {
+			counter: 0,
+			addScore: 1,
+			firstCount: 10,
+			firstAdd: 1,
+			secondAdd: 2,
+		};
+	}
+
+	increaseAmount = () => {
+		const { counter, addScore } = this.state;
+		this.setState({
+			counter: counter + addScore,
+		});
+	};
+	pointBoost = () => {
+		const { counter, firstCount, addScore, firstAdd, secondAdd } = this.state;
+		if (counter < firstCount) {
+			alert("You can't afford that!");
+		} else {
+			this.setState({
+				addScore: addScore + 1,
+				counter: counter - firstCount,
+				firstAdd: firstAdd + 1,
+				secondAdd: secondAdd + 1,
+			});
+		}
+	};
+	refresh = () => {
+		this.setState({
+			counter: 0,
+			addScore: 1,
+			firstCount: 10,
+			firstAdd: 1,
+			secondAdd: 2,
+		});
+	};
+
+	render() {
+		const { counter, addScore, firstAdd, secondAdd } = this.state;
```

Comment:
```suggestion
		const { counter, addScore } = this.state;
```
You only need `counter` and `addScore` in state. `firstAdd` and `secondAdd` are redundant. You can just use `addScore` and `addScore + 1` in place of `firstAdd` and `secondAdd` on line 61.